auslib/db.py

```diff
@@ -1140,6 +1140,13 @@ def update(self, where, what, changed_by, old_data_version, transaction=None, dr
             if not self._dataVersionsAreSynced(sc_id, transaction):
                 raise MismatchedDataVersionError("Conditions data version is out of sync with main table for sc_id %s" % sc_id)
 
+        if type(where) == dict and where.get("sc_id", None):
```

Comment:
Hmmm, I'm not sure this is going to work, especially for calls that use a list of SQLAlchemy expressions instead of a dict. I think you can probably make use of the "affected_ids" that are calculated above.